10-ConcurrencyAndSynchronization/mpu6050.c

```diff
@@ -195,10 +272,39 @@ CLASS_ATTR_RO(temperature);
 
 static struct class *attr_class;
 
+int bg_thread(void *data)
+{
+	pr_info("%s started", __func__);
+	while (!kthread_should_stop()) {
+		/**check requirement for*/
+		// pr_info("bg_thread check flag");
+		if (atomic_read(&update_req) != 0) {
+			pr_info("%s executing", __func__);
+			mpu6050_exec_reading();
+			atomic_set(&update_req, 0);
+			pr_info("reset update_req and read_done");
+			complete(&read_done);
+		}
+		schedule();
```

Comment:
Call to `schedule()` while in `TASK_RUNNING` state will only yield CPU but keeps task in runqueue.
So it will continue to waste CPU for context switching for polling `update_req` which is not optimal design.
It's better to put it into `TASK_INTERRUPTIBLE` state and wakup whenever required.
Or wait on some blocking API (semaphore, completion, etc.).